File: 프로그래머스/lv2/12951. JadenCase 문자열 만들기/JadenCase 문자열 만들기.java
import java.util.Arrays;
import java.util.stream.Collectors;

public class Main {
    public static void main(String[] args) {

        System.out.println(new Solution().solution("3people unFollowed me"));
        System.out.println(new Solution().solution("for the last week"));

    }
}

class Solution {
    public String solution(String s) {
        String[] arr = s.split(" ");

        String answer = Arrays.stream(arr)
                .map(e -> {
                    if (e.equals("")) {
                        return "";
                    } else {
                        return String.valueOf(e.charAt(0)).toUpperCase() + e.substring(1).toLowerCase();
                    }
                })
                .collect(Collectors.joining(" "));

        String xAnswer = String.valueOf(s.charAt(s.length() - 1));
        if (xAnswer.equals(" ")) {
            answer += " ";
        }
        return answer;
    }
}

File: 프로그래머스/lv2/12951. JadenCase 문자열 만들기/JadenCase 문자열 만들기.js
function solution(s) {
 let answer = [];
    let s_list = s.split(' '); // 공백을 기준으로 배열 생성
    s_list.forEach(function(val){ // s_list의 단어들을 꺼내오기
        answer.push(val.slice(0,1).toUpperCase()+val.slice(1).toLowerCase()); // 첫번째 글자는 대console.log(val)
    })
    return answer.join(' '); // 배열을 합친다
}

/*
if (!Character.isDigit(e.charAt(0))) {
                        return String.valueOf(e.charAt(0)).toUpperCase() + e.substring(1).toLowerCase();
                    }
                    return e.charAt(0) + e.substring(1).toLowerCase();
*/